src/cordova/android/OutlineAndroidLib/outline/src/main/java/org/outline/vpn/VpnTunnelService.java

```diff
@@ -621,17 +573,23 @@ public int getResourceId(final String name, final String type) {
     return getResources().getIdentifier(name, type, getPackageName());
   }
 
-  /* Returns the server's name from |serverConfig|. If the name is not present, it falls back to the
+  /* Returns the server's name from |proxyString|. If the name is not present, it falls back to the
    * host name (IP address), or the application name if neither can be retrieved. */
   private String getServerName(final TunnelConfig config) {
     try {
-      String serverName = config.name;
-      if (serverName == null || serverName.equals("")) {
-        serverName = config.proxy.host;
+      JSONObject object = (JSONObject) new JSONTokener(config.proxyString).nextValue();
```

Comment:
We should treat the proxy config as opaque. Move the name back to the TunnelConfig.